### PR TITLE
MM-53098 Fix for checking bot and user permissions on shared endpoints

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Update checking bot and user permissions on shared endpoints.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-53098

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by sbishel** [comment]
@esethna - We will need PM input on this feature. I'm not familiar enough with how this is supposed to work. We may need to create a spec as it appears there are several things that may need fixed.

File: server/channels/api4/user.go [hunk 1]
```diff
@@ -1335,6 +1335,12 @@ func patchUser(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.SetInvalidParam("user_id")
 		return
 	}
+	if ouser.IsBot {
```
Comment (hmhealey) [inline server/channels/api4/user.go:1338]: Should this maybe come after the `AddEventPriorState` below? That seems to fit with how the original user is included in the audit if one admin attempts to modify another

**Comment by esethna** [comment]
> We will need PM input on this feature. I'm not familiar enough with how this is supposed to work. We may need to create a spec as it appears there are several things that may need fixed.

@sbishel @esarafianou I think I'm missing some context on this. What decisions are we blocked on? We can have the conversation in MM as needed

**Review by hmhealey — APPROVED** [review]
Code looks good on my end, but since I'm not involved in that conversation, feel free to request a re-review if anything changes substantially

**Review by esarafianou — APPROVED** [review]

**Pull request merged**